presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -56,6 +56,7 @@ statement
         DROP COLUMN column=qualifiedName                               #dropColumn
     | ALTER TABLE tableName=qualifiedName
         ADD COLUMN column=columnDefinition                             #addColumn
+    | ANALYZE qualifiedName (WITH properties)?                         #analyze
```

Comment:
Do you plan on having first-class support for selecting columns to analyze?
eg `ANALYZE table [ ( column, column, ... ) ] [ WITH ... ]`

Reply:
The idea was to have a special mechanisms that allows to mark what columns need statistics, and what statistics are needed for that columns. These markers could be set automatically (by instrumenting optimizer), or manually, with an additional SQL statement.

Reply:
we don't have this instrumentation. Also, it would work only for sustainable workloads. Sometimes, I know my data better than my cluster, because cluster was just started...
if a user is allowed to select columns to ANALYZE, they should be able to do it in the most natural way. It's not a problem if it's not supported in v1 though.

Reply:
We had a long discussion about that. Selecting particular columns is not enough. Some of the statistics are needed on every column (like data size or null count), when other statistics, like NDV, which is heavy, may not be needed for every column. Once we have histogramms and maybe correlations - it will become even more problem. And speaking of correlations - for correlations you need to specify 2 columns, no a singe column.


The syntax to express everything of the above will be very complex. That's why we thought it could be better idea to have a separate statement that will ENABLE/DISABLE statistics to collect per column.

Reply:
Anyhow, let's leave v1 as is. And than we can discuss it.

Reply:
Thanks for sharing this. Indeed enlightening and I agree, we will need separate statements (or equivalent) for fine tuning.
OTOH, when a user wants something simple ("i want to analyze the columns used as filters/conditions in the queries"), they don't want to hear argumentation about elaborated features they don't (yet) need. Thus, i think we should consider doing both.